mlrun/api/launcher.py

```diff
@@ -147,8 +148,14 @@ def verify_base_image(runtime):
         pass
 
     @staticmethod
-    def _enrich_runtime(runtime):
-        pass
+    def _enrich_runtime(runtime: "mlrun.runtimes.base.BaseRuntime"):
+        if runtime.metadata.project:
```

Comment:
in what cases runtime is not a part of a project in the API?

Reply:
runtimes are always executed as part of a project, you might need to set the project name prior to enriching the runtime or pass it if its not already been done.

Reply:
added project enrichment